### PR TITLE
Use modreq to prevent consumption of instance extensions from other compilers

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs [hunk 1]
```diff
@@ -44143,7 +44229,17 @@ static void Main()
...
+            // (6,11): error CS0570: 'E.add_E1(ref C, Action)' is not supported by the language
+            //         E.add_E1(ref c, M2());
+            Diagnostic(ErrorCode.ERR_BindToBogus, "add_E1").WithArguments("E.add_E1(ref C, System.Action)").WithLocation(6, 11),
+            // (8,11): error CS0570: 'E.add_E1(ref C, Action)' is not supported by the language
+            //         E.add_E1(ref c, M3());
+            Diagnostic(ErrorCode.ERR_BindToBogus, "add_E1").WithArguments("E.add_E1(ref C, System.Action)").WithLocation(8, 11),
+            // (9,11): error CS0570: 'E.remove_E1(ref C, Action)' is not supported by the language
+            //         E.remove_E1(ref c, M2());
+            Diagnostic(ErrorCode.ERR_BindToBogus, "remove_E1").WithArguments("E.remove_E1(ref C, System.Action)").WithLocation(9, 11)
+            );
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs:44242]: nit: consider adding a test with a `modopt([System.Runtime]System.Runtime.CompilerServices.ExtensionAttribute)`

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

File: src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEPropertySymbol.cs [hunk 2]
```diff
@@ -289,13 +300,29 @@ public bool TryGetHasUnscopedRefAttribute(out bool hasUnscopedRefAttribute)
...
+                for (int i = 0; i < propertyParams.Length; i++)
+                {
+                    var p = propertyParams[i];
+
+                    if (!p.RefCustomModifiers.IsDefaultOrEmpty && p.RefCustomModifiers.Any(static m => !m.IsOptional && !m.Modifier.IsWellKnownTypeInAttribute()))
+                    {
+                        return true;
+                    }
+
+                    if ((!isExtensionInstanceUnderlyingSymbol || i != 1) && p.CustomModifiers.AnyRequired())
```
Comment (jjonescz) [inline src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEPropertySymbol.cs:314]: Should it be `i != 0`?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEPropertySymbol.cs:314]: > Should it be `i != 0`?

No, an information for return type comes at position 0.

**Review by jjonescz — APPROVED** [review]

**Pull request merged**